### PR TITLE
UCT/CUDA_COPY: Move to using non-blocking stream for get_short

### PR DESCRIPTION
**Pull request by Akshay-Venkatesh**
- Move away from using cudaMemcpy on default stream for `get_short` as well.

@bureddy Can you review?

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10242/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10280/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
```
15:35:55 [     INFO ] memory_type:cuda 0 4294967295 /scrap/jenkins/workspace/hpc-ucx-pr-3/label/vulcan-jenkins/worker/0/contrib/../test/gtest/uct/uct_test.cc:1060: Failure
15:36:11 At offset 0: Expected: 0x2222222222222222 Got: 0x1111111111111111
15:36:11 
15:36:11 
[  FAILED  ] cuda_copy/uct_p2p_rma_test.get_short/1, where GetParam() = cuda_copy/cudacopy0/loopback (16036 ms)
```

File: src/uct/cuda/cuda_copy/cuda_copy_ep.c [hunk 1]
```diff
@@ -149,10 +149,14 @@ ucs_status_t uct_cuda_copy_ep_get_short(uct_ep_h tl_ep, void *buffer,
...
 {
+    uct_cuda_copy_iface_t *iface = ucs_derived_of(tl_ep->iface, uct_cuda_copy_iface_t);
     ucs_status_t status;
 
-    status = UCT_CUDA_FUNC(cudaMemcpy(buffer, (void *)remote_addr,
-                                      length, cudaMemcpyDeviceToHost));
+    UCT_CUDA_COPY_CHECK_AND_CREATE_STREAM(iface->stream_d2h);
+
+    UCT_CUDA_FUNC(cudaMemcpyAsync((void *)remote_addr, buffer, length,
+                                  cudaMemcpyHostToDevice, iface->stream_d2h));
```
Comment (bureddy) [inline src/uct/cuda/cuda_copy/cuda_copy_ep.c]: @Akshay-Venkatesh used wrong direction (cudaMemcpyHostToDevice) and src, dst parms

Comment (Akshay-Venkatesh) [reply src/uct/cuda/cuda_copy/cuda_copy_ep.c]: shoot! Fixing this now

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10301/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10302/ for details (Mellanox internal link).

**Pull request merged**